core/src/main/java/feast/core/model/FeatureStatistics.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.model;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.io.*;
+import java.util.Date;
+import java.util.List;
+import javax.persistence.*;
+import javax.persistence.Entity;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.tensorflow.metadata.v0.*;
+
+@NoArgsConstructor
+@Getter
+@Setter
+@Entity
+@Table(
+    name = "feature_statistics",
+    indexes = {
+      @Index(
+          name = "idx_feature_statistics_feature",
+          columnList = "project,feature_set,version,name"),
+      @Index(name = "idx_feature_statistics_dataset_id", columnList = "datasetId"),
+      @Index(name = "idx_feature_statistics_date", columnList = "date"),
+    })
+public class FeatureStatistics {
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private int id;
+
+  @ManyToOne
+  @JoinColumns({
+    @JoinColumn(name = "project", referencedColumnName = "project"),
+    @JoinColumn(name = "feature_set", referencedColumnName = "feature_set"),
+    @JoinColumn(name = "version", referencedColumnName = "version"),
+    @JoinColumn(name = "name", referencedColumnName = "name")
+  })
+  private Feature feature;
+
+  // Only one of these fields should be populated.
+  private String datasetId;
+  private Date date;
+
+  // General statistics
+  private String featureType;
+  private long count;
+  private long numMissing;
+  private long minNumValues;
+  private long maxNumValues;
+  private float avgNumValues;
+  private long totalNumValues;
+  private byte[] numValuesHistogram;
+
+  // Numeric statistics
+  private double mean;
+  private double stdev;
+  private long zeroes;
+  private double min;
+  private double max;
+  private double median;
+  private byte[] numericValueHistogram;
+  private byte[] numericValueQuantiles;
+
+  // String statistics
+  @Column(name = "n_unique")
+  private long unique;
+
+  private float averageLength;
+  private byte[] rankHistogram;
+  private byte[] topValues;
+
+  // Byte statistics
+  private float minBytes;
+  private float maxBytes;
+  private float avgBytes;
+
+  // Instantiates a Statistics object from a tensorflow metadata FeatureNameStatistics object and a
+  // dataset ID.
+  public static FeatureStatistics fromProto(
```

Comment:
I think we should perhaps change the method names in this class. `fromProto` and `toProto` are normally reserved for `domain model` conversion to and from our proto representation. We are passing non-proto methods here.

Reply:
`FeatureNameStatistics` is a proto.